graphql/resolvers.go

```diff
@@ -184,6 +184,20 @@ func (r *volumeResolver) Host(ctx context.Context, obj *restModel.APIVolume) (*r
 	return &apiHost, nil
 }
 
+func (r *queryResolver) IsPatchOrVersion(ctx context.Context, id string) (PatchOrVersionType, error) {
```

Comment:
Could this instead be HasPatch and return a boolean or something? Technically patches can have versions and the ID would work for both; also "PatchOrVersionTypePatch" is wordy. Don't have the full use case picture though.

Reply:
I prefer this approach as well!

Reply:
The way i imagined this working was the following.
Client sends the id to `IsPatchOrVersion` 
GQL returns `PATCH` or `VERSION` or `ERROR`
- if `PATCH` is returned 
  1) We can safely request data about the patch and send out an initial request which returns data about if the patch is configured (And has a version) and its commit queue status. 
     - if the Patch is configured and has a Version we know we can safely request the version data that we need to render the page
     - If the patch is on the commit queue but not yet running we can redirect to the commit queue page
     - If the Patch is not configured we can redirect to the configure page
- If `VERSION` is returned 
  1) We can safely request the necessary version data we need to render the page since we know the version exists
- if `ERROR` is returned
  1) We can throw an error because we know there is no version or patch in evergreen with the associated ID and we don't have to send out any subsequent requests to determine this at a later stage in loading the page.

Reply:
`PatchOrVersionTypePatch` This was autogenerated by gqlgen.

Reply:
I guess I'm still not sure why this can't be a boolean instead. "IsPatchOrVersion" just feels misleading; patches can also have versions. 

If "the function was "HasVersion": 
* If `TRUE` is returned, do the "is version is returned" step.
* If `FALSE` is returned, do the "is patch is returned" step (because if it doesn't have a version then it does have a patch).
* If `ERROR` is returned (which we should return if we can't validate the patch / version thing): do the error step.

This eliminates the need for "PatchOrVersionType" at all.

Reply:
Great point! I didn't consider that. Just made that change.